File: DSA/twosum.js



function twoSum(arr,k){

  for (let i=0; i<arr.length; i++){
    for (let j=i+1; j<arr.length; j++){
      if (arr[i] + arr[j] == k){
        console.log(arr[i],arr[j])
        return 
      }
    }
  }
  console.log("Not getting any pair ")
}

// twoSum([1,2,3,4,5],6)

function twoPointer(arr,k){
  arr.sort((a,b)=> a-b);
  // console.log(arr)
  let left = 0;
  let right = arr.length-1;
  while (left < right){
    let sum = arr[left] + arr[right] 
    if (sum == k){
      console.log(left,right)
      return 
    }
    else if (sum > k){
      right --
    }
    else{
      left ++
    }
  }
  return console.log("not getting any pair")
}
twoPointer([1,10,3,5,4],8)

File: Python/twoSum.py
def twosum(arr, target):
    n = len(arr)
    for i in range(n):
        for j in range(i + 1, n): 
            if arr[i] + arr[j] == target:
                print("True")
                return
    print("False")  # also fix this: should print instead of returning string

def twopointer(arr,target):
    arr.sort()
    print(arr)
    n = len(arr)
    l = 0
    r = n-1

    while l < r:
        currentsum = arr[l] + arr[r]
        if currentsum == target:
            print("true")
            print(l,r)
            return
        elif currentsum < target:
            l += 1
        else:
            r -= 1
    print("false")
arr = [2,3,1,5,6]
target = 7
twosum(arr,target)
twopointer(arr,target)